worker/caasunitprovisioner/deployment_worker.go

```diff
@@ -146,7 +146,13 @@ func (w *deploymentWorker) loop() error {
 		if err != nil {
 			return errors.Annotate(err, "cannot parse pod spec")
 		}
-
+		if len(spec.CustomResourceDefinitions) > 0 {
```

Comment:
This should be internal to EnsureService().
The interface we expose on the caas client reflects generic juju-like semantics (service, units etc) and doesn't expose substrate specific functionality. The idea is that we may at some point support something other than k8s (but we probably never will).

Reply:
As discussed on IRC, charms could define custom resources without pods.
We need to allow this but also need to re-think how we model things

Reply:
yes, agreed.